src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -35,6 +36,75 @@ const std::string MFEMSidreDataCollection::s_attribute_suffix =
   "_material_attribute";
 const std::string MFEMSidreDataCollection::s_coordset_name = "coords";
 
+namespace detail
+{
+/**
+ * @brief Retrieves the last "n" tokens of a string split with the specified delimiter
+ * @param[in] input The string to split
+ * @param[in] n The number of tokens to retrieve
+ * @param[in] delim The delimiter to split with
+ * 
+ * @return A list of tokens (of size @p n )
+ * 
+ * Splits a string starting from the end of the string into a maximum of @p n tokens
+ */
+std::vector<std::string> splitLastNTokens(const std::string& input,
+                                          const std::size_t n,
+                                          const char delim)
+{
+  std::vector<std::string> result;
+
+  auto last_pos = std::string::npos;
+  auto pos = input.find_last_of(delim, last_pos - 1);
+
+  while((pos != std::string::npos) && (result.size() < n - 1))
+  {
+    result.push_back(input.substr(pos + 1, last_pos - pos - 1));
+    last_pos = pos;
+    pos = input.find_last_of(delim, last_pos - 1);
+  }
+  // Add the rest of the string (first token)
+  result.push_back(input.substr(0, last_pos));
+  std::reverse(result.begin(), result.end());
+  return result;
+}
+
+/**
+ * @brief Implements an analogue to mfem::FiniteElementCollection::New
+ * for mfem::QuadratureSpaces - basis currently must be of form QF_Default_[ORDER]_[VDIM]
+ * @param[in] name The name that encodes the QuadratureSpace data
+ * @param[in] mesh The mesh to construct the QuadratureSpace with
+ * @param[out] vdim The vector dimension parsed from the basis name - unlike FEColl/FESpace,
+ * QSpaces do not contain vdim information, but it may be useful for the caller when constructing
+ * the QFunc
```

Comment:
Note that the `vdim` for an `mfem::QuadratureFunction` is stored in the `mfem::QuadratureFunction` itself and not in the `mfem::QuadratureSpace`. This way different `mfem::QuadratureFunction`s with different `vdim` can use the same `mfem::QuadratureSpace`.

This is in contrast to `mfem::(Par)GridFunction` where `vdim` is part of the `mfem::(Par)FiniteElementSpace`.

Reply:
@v-dobrev so for the basis string I go over the logic a bit here: https://github.com/LLNL/axom/pull/433#discussion_r577936473 . So, the string itself encodes information related to the quadrature space and quadrature function vdim. I believe the logic here for this function is that since you're already parsing the basis string to reconstruct the `QuadratureSpace` you ought as well return what the VDim is for the `QF`  to the user in case the user wants it.

Reply:
Okay, I just wanted to point out this small difference in handling `vdim` in `mfem::QuadratureFunction` compared to `mfem::(Par)GridFunction`. I guess you already new that. :-)